### PR TITLE
chore: Small API doc fixes

### PR DESCRIPTION
**Pull request by nurupo**
API docs are a bit inconsistent:

1. sometimes use "tox" instead of "Tox"
2. sometimes use "null" instead of "NULL"
3. sometimes calls strings NUL-terminated, other times NULL terminated
4. sometimes comments go over 80 columns
5. sometimes `@param` are listed after `@return`
6. sometimes it's a function, other times it's a \`function\`, and yet other times it's a \`function()\`, same for parameters or \`parameters\` -- oh, and sometimes a single comment mixes multiple styles! (e.g. tox_group_get_topic)
7. NGC functions doesn't use `@breif` for some reason when all other functions do. Why?
8. some functions don't document their parameters, e.g. tox_group_set_topic, while others document even the obvious most parameters.

This PR addresses only the first 4 points, and one case of the 5th point that I have noticed by an accident.

5-8 points are not addressed. Would be nice if someone could address those.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2735)

### CODE REVIEW

File: toxcore/tox.h [hunk 1]
```diff
@@ -473,7 +465,8 @@ typedef enum Tox_Log_Level {
     TOX_LOG_LEVEL_WARNING,
 
     /**
-     * Severe unexpected errors caused by external or events_alloc inconsistency.
+     * Severe unexpected errors caused by external or events_alloc
```
Comment (iphydf) [inline toxcore/tox.h:468]: events_alloc should be "internal". This was a mistake made by me and not caught in review.

File: toxcore/tox.h [hunk 2]
```diff
@@ -482,7 +475,8 @@ typedef enum Tox_Log_Level {
 const char *tox_log_level_to_string(Tox_Log_Level value);
 
 /**
- * @brief This event is triggered when the toxcore library logs an events_alloc message.
+ * @brief This event is triggered when the toxcore library logs an events_alloc
```
Comment (iphydf) [inline toxcore/tox.h:478]: Here also, "internal".

File: toxcore/tox.h [hunk 3]
```diff
@@ -837,15 +834,16 @@ typedef enum Tox_Err_New {
     TOX_ERR_NEW_NULL,
 
     /**
-     * The function was unable to allocate enough memory to store the events_alloc
-     * structures for the Tox object.
+     * The function was unable to allocate enough memory to store the
+     * events_alloc structures for the Tox object.
```
Comment (iphydf) [inline toxcore/tox.h:838]: "internal", maybe check the rest of the file(s) for having that mistake.

File: toxcore/tox.h [hunk 4]
```diff
@@ -1065,14 +1064,16 @@ typedef void tox_self_connection_status_cb(Tox *tox, Tox_Connection connection_s
...
+ * @brief Return the time in milliseconds before `tox_iterate()` should be
+ *   called again for optimal performance.
  */
 uint32_t tox_iteration_interval(const Tox *tox);
 
 /**
- * @brief The main loop that needs to be run in intervals of `tox_iteration_interval()`
- *   milliseconds.
+ * @brief The main loop that needs to be run in intervals of
+ * `tox_iteration_interval()` milliseconds.
```
Comment (iphydf) [inline toxcore/tox.h:1074]: Is there a reason not to indent the `@brief`?

File: toxcore/tox.h [hunk 5]
```diff
@@ -2234,15 +2252,16 @@ const char *tox_err_file_send_to_string(Tox_Err_File_Send value);
  * - If the file size was increased
  *   - and sending mode was streaming (file_size = UINT64_MAX), the behaviour
  *     will be as expected.
- *   - and sending mode was file (file_size != UINT64_MAX), the file_chunk_request
- *     callback will receive length = 0 when Core thinks the file transfer has
- *     finished. If the client remembers the file size as it was when sending the
- *     request, it will terminate the transfer normally. If the client re-reads the
- *     size, it will think the friend cancelled the transfer.
+ *   - and sending mode was file (file_size != UINT64_MAX), the
+ *     file_chunk_request callback will receive length = 0 when Core thinks the
```
Comment (iphydf) [inline toxcore/tox.h:2256]: Maybe "Tox" instead of "Core"?

**Review by iphydf — CHANGES_REQUESTED** [review]

**Comment by nurupo** [comment]
@JFreegman What does it mean when NGC says "core error"? I'm confused by it being named "core". Does that mean it's a toxcore error? Or perhaps it means "a fatal error" -- something horribly went wrong, without specifying what it is, like some sort of a general error? Maybe it's better not to call it a core error but to come up with some other synonym / wording?

https://github.com/TokTok/c-toxcore/blob/3e05824b80eb9bee33e8254cba0780d84c522182/toxcore/tox.h#L3641-L3644

https://github.com/TokTok/c-toxcore/blob/3e05824b80eb9bee33e8254cba0780d84c522182/toxcore/tox.h#L3738-L3741

https://github.com/TokTok/c-toxcore/blob/3e05824b80eb9bee33e8254cba0780d84c522182/toxcore/tox.h#L4890-L4893

File: toxcore/tox.h (hunk 5 above)
Comment (nurupo) [reply toxcore/tox.h:2256]: Some parts of the API refer to the library as "Tox" and "Tox library", other as "toxcore" and "the toxcore library", other yet as "Core"...

**Comment by nurupo** [comment]
@iphydf `const Tox_System *operating_system` is located under the experimental comment disclaimer, sandwiched between two `experimental_*` options, yet it is not marked as experimental, i.e. it's not named `experimental_operating_system`. Does that mean that it's not experimental and a part of the supported API, as per the experimental comment disclaimer? Or did we just forget to prefix its name with `experimental_`?

https://github.com/TokTok/c-toxcore/blob/3e05824b80eb9bee33e8254cba0780d84c522182/toxcore/tox.h#L662-L692

File: toxcore/tox.h (hunk 5 above)
Comment (nurupo) [reply toxcore/tox.h:2254]: I have tried addressing this in a separate commit, please check.

File: toxcore/tox.h (hunk 4 above)
Comment (nurupo) [reply toxcore/tox.h]: Missed that, thanks. Should be fixed now, fixedup/squashed into the original commit.

File: toxcore/tox.h (hunk 3 above)
Comment (nurupo) [reply toxcore/tox.h]: Should be fixed now, please double check.

File: toxcore/tox.h (hunk 2 above)
Comment (nurupo) [reply toxcore/tox.h]: Should be fixed now, please check.

File: toxcore/tox.h (hunk 1 above)
Comment (nurupo) [reply toxcore/tox.h]: Should be fixed now, please check.

**Comment by iphydf** [comment]
operating_system is experimental.

**Comment by nurupo** [comment]
@iphydf well, it's not obvious.

>      * These options are experimental, so avoid writing code that depends on 
>      * them. Options marked "experimental" may change their behaviour or go away 
>      * entirely in the future, or may be renamed to something non-experimental 
>      * if they become part of the supported API. 

I don't see it being marked as experimental in any way.

**Comment by JFreegman** [comment]
> What does it mean when NGC says "core error"? I'm confused by it being named "core". Does that mean it's a toxcore error? Or perhaps it means "a fatal error" -- something horribly went wrong, without specifying what it is, like some sort of a general error? Maybe it's better not to call it a core error but to come up with some other synonym / wording?

In the case of join and rejoin, it means Messenger.c failed to create a friend connection for some reason. In the case of invite accept, see: https://github.com/TokTok/c-toxcore/pull/2736

File: toxcore/tox.h (hunk 5 above)
Comment (nurupo) [reply toxcore/tox.h:2254]: Oh, I actually missed this particular instance of "Core". Let me update it.

Comment (nurupo) [reply toxcore/tox.h]: There you go :)

**Comment by nurupo** [comment]
@JFreegman that's one, what about the other two core errors? No plan to rename them to "initiation", "general", "fatal" or anything else, keeping them as "core" errors?

**Comment by JFreegman** [comment]
> @JFreegman that's one, what about the other two core errors? No plan to rename them to "initiation", "general", "fatal" or anything else, keeping them as "core" errors?

I'm sort of indifferent. I think "core" conveys about as much information as anything else. You can change it if you want.

File: toxcore/tox.h [hunk 6]
```diff
@@ -250,7 +241,8 @@ uint32_t tox_nospam_size(void);
  * @brief The size of a Tox address in bytes.
  *
  * Tox addresses are in the format
- * `[Public Key (TOX_PUBLIC_KEY_SIZE bytes)][nospam (4 bytes)][checksum (2 bytes)]`.
+ * `[Public Key (TOX_PUBLIC_KEY_SIZE bytes)][nospam (4 bytes)]
```
Comment (nurupo) [inline toxcore/tox.h:244]: @iphydf cimple errors on this line

```
Processing toxcore/tox.h
toxcore/tox.h:244:4: Parse error near lexical error: "`"; expected end of file
Error: Process completed with exit code 1.
```

Comment (iphydf) [reply toxcore/tox.h:244]: It doesn't expect ` to run beyond a single line.

Comment (nurupo) [reply toxcore/tox.h:244]: Want me to revert it to being a single line, or do we want cimple to learn matching ` across lines?

Comment (iphydf) [reply toxcore/tox.h:244]: Revert for now, I don't have time to teach cimple this week.

Comment (nurupo) [reply toxcore/tox.h]: Done

**Comment by JFreegman** [comment]
Is there a reason we're putting this in the .19 release? Documentation cleanup isn't critical for the release is it?

**Comment by nurupo** [comment]
@JFreegman The very first paragraphs of tox.h talk about apidsl, which we no longer use, about `tox.in.h`, which doesn't exist, about old style callbacks, which also don't exist anymore. It makes sense to correct these points before a release. That's what the very first commit addresses, and what the PR was going to be about initially. All the following commits are mostly cleanups of things that I noticed after making the first commit rather than being corrections. Aside perhaps from iphy forgetting to s/events_alloc/internal/ in tox.h, which is another correction this PR makes that was discovered later. So anyway, it's not just cleanups and I think it makes sense for .19.

The PR is ready to be merged. It looks like we are waiting on @iphydf to mark his comments as resolved and that's it. Any day now. Weird how he approved the PR but didn't resolve them. Or perhaps he expects me to mark them as resolved?

**Comment by JFreegman** [comment]
> Or perhaps he expects me to mark them as resolved?

I usually mark them as resolved myself when I resolve them.

**Comment by iphydf** [comment]
Yes please resolve it for me. I'm on the road and don't have a laptop so I'm doing this all on the phone, where it's hard to find that comment. I approved it, it looks good to me, no comments need addressing.

**Pull request merged**